gems/sorbet-runtime/lib/types/props/type_validation.rb

```diff
@@ -42,7 +42,7 @@ def prop_validate_definition!(name, _cls, rules, type)
       )
       .void
     end
-    private def validate_type(type, field_name:)
+    private def validate_type(type, field_name)
```

Comment:
If we're saying that type checking this method is hot, should we mark the sig `.checked(:never)`?

Reply:
Yes, we can. I guess I wasn't sure if we wanted to lose whatever guarantees we get here from the checking, but the likelihood is that a sig is checked somewhere well up the callstack, so it's ok to make this checked-never.